### ISSUE TITLE
How to iterate over training set AGAIN on training epoch end?

### ISSUE THREAD

**Issue by noamzilo** [PyTorchLightning/pytorch-lightning#5552]
#### What is your question?

How to iterate over training set AGAIN on training epoch end or on validation epoch start?
I have a model that works as expected on MNIST, but for my unique data, val_loss<train_loss for all samples.
I have no idea what causes this, and it is too suspicious to allow me to go on.

*I want to do a validation step on training data, in eval mode*

I hope it will ease my mind.

#### Code
Well, if I had code for how to correctly do this I wouldn't ask :)

#### What have you tried?
This doesn't sound like a standard use case, not even sure that's supported.

#### What's your environment?

 - OS: [e.g. iOS, Linux, Win] Win
 - Packaging [e.g. pip, conda] Pip
 - Version [e.g. 0.5.2.1] 1.1.4

**Comment by tchaton** [comment]
Hey @noamzilo,

If you have a `val_dataloader`, the entire dataset of validation will be used once the latest batch of your train_dataset would be reached.

Therefore, you can compare your metrics at this point in epoch_end hooks.

Best,
T.C

**Comment by awaelchli** [comment]
It is not an accident that PyTorch is in the name of PyTorchLightning :)) 
You can just do it as you would in PyTorch:

```python
self.eval()
for idx, batch in enumerate(self.train_dataloader()):
    # do what you have to do
```

**Comment by noamzilo** [comment]
> 
> 
> It is not an accident that PyTorch is in the name of PyTorchLightning :))
> You can just do it as you would in PyTorch:
> 
> ```python
> self.eval()
> for idx, batch in enumerate(self.train_dataloader()):
>     # do what you have to do
> ```

Thanks :)
